packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1054,6 +1068,62 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   * Updates, re-authorizes and then restarts given snap.
+   *
+   * @param snapId The id of the Snap to be updated
+   * @param newVersionRange A semver version range in which the maximum version will be chosen
+   * @returns @type {TruncatedSnap} if updated, @type {null} otherwise
+   */
+  async updateSnap(
+    snapId: SnapId,
+    newVersionRange: string = DEFAULT_REQUESTED_SNAP_VERSION,
+  ): Promise<TruncatedSnap | null> {
+    const snap = this.get(snapId);
+    if (snap === undefined) {
+      throw new Error(
+        `Could not find snap ${snapId}. Install the snap before attempting to update it.`,
+      );
+    }
+
+    const newSnap = await this._fetchSnap(
```

Comment:
Although the logic of this function is sound either way, we should check whether the the requested version _range_ is valid before fetching the snap. Fetching a snap may be expensive, and I think this optimization would be worth it. I'd say that we could do this in the RPC method implementation, but we may want to call this method from within MetaMask itself, in which case we'd have to implement that check in multiple places.

Reply:
It's already checked in `_fetchNpmSnap` but I'll add a check anyways